go/enclave/rollupchain/rollup_chain.go

```diff
@@ -270,6 +261,8 @@ func (rc *RollupChain) updateState(b *types.Block) (*obscurocore.BlockState, map
 		subscribedLogs[subscriptionID] = logsForID
 	}
 
+	rc.storage.SaveNewHead(bs, head, receipts, subscribedLogs)
```

Comment:
Feels like we should store the logs *before* we append the parent logs but I guess not if we want the full growing history.

Trying to convince myself this never results in us accidentally appending the same ancestor logs multiple times in the same log history, and think it's fine but one to watch out for if it spirals out of control faster than expected maybe.